scripts/menus/options.lua

```diff
@@ -391,15 +391,35 @@ function RunPreferencesMenu()
          SetFogOfWarType(fogOfWarTypes[fogOfWarType:getSelected() + 1])
    end)
    fogOfWarType:setSize(120, 16)
+
+   local maxSelectableUnits = {9, 12, 18, 50, 100, 200}
+   local maxSelectableUnitsList = {"9", "12", "18", "50", "100", "200"}
```

Comment:
Sad to duplicate list.
int<->string conversion can be done on the fly IMO.

Textfield/spinbox is another option. Not sure we have validator though.

Reply:
The conversion from int to string and vice versa is easy, that's true.
I was thinking this change
```lua
local maxSelectableUnitsList = {"9", "12", "18", "50", "100", "200"}
   menu:addLabel(_("Selectable units:"),  225, 28 + 19 * 8 + 5, Fonts["game"], false)
   local maxSelectableUnitsListDD = menu:addDropDown(maxSelectableUnitsList, 225, 28 + 19 * 9 + 5, function(dd) end)
   -- Get the index of the MaxSelectableUnits from list
   local idx_msu = 0
   for i,v in ipairs(maxSelectableUnitsList) do
      if tonumber(v) == wc2.preferences.MaxSelectableUnits then
         idx_msu = i - 1
         break
      end
   end
   maxSelectableUnitsListDD:setSelected(idx_msu)
   maxSelectableUnitsListDD:setActionCallback(
      function()
         Preference.MaxSelectableUnits = tonumber(maxSelectableUnitsListDD:getSelected())
      end
   )
   maxSelectableUnitsListDD:setSize(40, 16)
```
The problem I have is the dropDown function `getSelected()` doesn't return the value from the list but the index of it. For example if I select 18 the function will return 2 and not 18.
I cannot find the function that actually returns the value I want and not the index from it.

Reply:
`ImageDropDownWidget` has `std::string getSelectedItem()`.
Else capture the table `maxSelectableUnitsList`

Reply:
I forgot to mention that I did already test that
```
maxSelectableUnitsListDD:setActionCallback(
      function()
         Preference.MaxSelectableUnits = maxSelectableUnitsListDD:getSelectedItem()
      end
   )
```
But I got this error
`[string "/home/not_important/.stratagus/data.Wargus/scripts..."]:410: attempt to call method 'getSelectedItem' (a nil value)`

Is this function added recently? 
Because I'm testing the changes on the stable version (3.3.2)

Reply:
By the way I have solved with this code:
```lua
maxSelectableUnitsListDD:setActionCallback(
      function()
         Preference.MaxSelectableUnits = tonumber(maxSelectableUnitsList[maxSelectableUnitsListDD:getSelected() + 1])
      end
   )
```